Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,512 @@
+        addSubviewWithConstraints(arSCNView)
+        arSCNView.delegate = self
+        
+        // Add sceneView to view and setup constraints.
+        addSubviewWithConstraints(sceneView)
+
+        // Make our sceneView's background transparent, no atmosphereEffect.
+        sceneView.isBackgroundTransparent = true
+        sceneView.atmosphereEffect = .none
+        
+        // Tell the sceneView we will be calling `renderFrame()` manually.
+        sceneView.isManualRendering = true
+        
+        // We haven't yet notified user of start or failure.
+        notifiedStartOrFailure = false
+        
+        // Set intitial orientationQuat.
+        orientationChanged(notification: nil)
+    }
+    
+    // MARK: Public
+    
+    /// Determines the map point for the given screen point.
+    ///
+    /// - Parameter toLocation: The point in screen coordinates.
+    /// - Returns: The map point corresponding to screenPoint.
+    public func arScreenToLocation(screenPoint: AGSPoint) -> AGSPoint {
+        fatalError("arScreen(toLocation:) has not been implemented")
+    }
+    
+    /// Resets the device tracking, using `originCamera` if it's not nil or the device's GPS location via the locationManager.
+    public func resetTracking() {
+        initialLocation = nil
+        startTracking()
+    }
+
+    /// Resets the device tracking, using the device's GPS location via the locationManager.
+    ///
+    /// - Returns: Reset operation success or failure.
+    public func resetUsingLocationServices() -> Bool {
+        fatalError("resetUsingLocationServices() has not been implemented")
+    }
+    
+    /// Resets the device tracking using a spacial anchor.
+    ///
+    /// - Returns: Reset operation success or failure.
+    public func resetUsingSpatialAnchor() -> Bool {
+        fatalError("resetUsingSpatialAnchor() has not been implemented")
+    }
+    
+    /// Starts device tracking.
+    public func startTracking() {
+        if !isSupported {
+            didStartOrFailWithError(ARError(.unsupportedConfiguration))
+            return
+        }
+        
+        if let origin = originCamera {
+            // We have a starting camera.
+            initialTransformationMatrix = origin.transformationMatrix
+            sceneView.setViewpointCamera(origin)
+            finalizeStart()
+        }
+        else {
+            // No starting camera, use location manger to get initial location.
+            let authStatus = CLLocationManager.authorizationStatus()
+            switch authStatus {
+            case .notDetermined:
+                startWithAccessNotDetermined()
+            case .restricted, .denied:
+                startWithAccessDenied()
+            case .authorizedAlways, .authorizedWhenInUse:
+                startWithAccessAuthorized()
+            }
+        }
+        
+        //TODO:  reloook at the mechanism by which we're getting notified of orientation changed events...
```

Comment:
relook

Reply:
I removed that comment; I took another look at it after Phil's review and I'm comfortable with how it works.